app/__init__.py

```diff
@@ -158,6 +158,15 @@ def create_app(application, config=None):
     return application
 
 
+def register_notify_blueprint(application, blueprint, auth_function, prefix=None):
+    if not blueprint._got_registered_once:
+        blueprint.before_request(auth_function)
+        if prefix:
+            application.register_blueprint(blueprint, url_prefix=prefix)
+        else:
+            application.register_blueprint(blueprint)
```

Comment:
So before if a blueprint was already registered and you called `before_request()` on it, the `before_request()` would basically be ignored. Now it's an explicit error (which is probably better than just ignoring people's code).

The app doesn't care, this code only runs once on boot, but the tests run this repeatedly and, essentially, the second test that got run would fail with the error.